client/src/game/gameHandelers/GameHandler.js

```diff
@@ -89,33 +90,46 @@ class GameHandler {
         }
     }
 
-    async timerP(duration) {
-        let waitP = (sec) => new Promise(resolve => setTimeout(resolve, sec*1000))
-        this._game.time = duration
-        while(duration >= 0) {
-            await waitP(1)
-            if (this._game.gameTime == 0) break
-            duration = duration - 1
-            this._game.time = duration
-        }
-        this._game.time = null
+    async startTimer(duration) {
+        return new Promise(async (resolve) => {
+            this._sw.start(count => {
+                this._game.time = duration - count > 0 ? duration - count : 0
+                if (duration - count <= 0) {
+                    this._sw.stop()
+                    resolve()
+                }
+            })
+        })
+    }
+
+    stopTimer() {
+        this._sw.stop()
     }
 
-    async playP() {
+    async preparePhaseP() {
         let waitP = (sec) => new Promise(resolve => setTimeout(resolve, sec*1000))
         await waitP(0.5)
         this._game.setRole(this._player.id, 'myCard')
-        await this.timerP(5) // give players a chance to internalize their card
-        console.log('night!')
+        await this.startTimer(5) // give players a chance to internalize their card
+    }
+
+    async nightPhaseP() {
+        let waitP = (sec) => new Promise(resolve => setTimeout(resolve, sec*1000))
+        
         this._status = 'night'
+        console.log('night!')
         this._startNightP()
-        await this.timerP(30) // give players a chance to perform their action
+        await this.startTimer(30) // give players a chance to perform their action
         while(this._midClick) await waitP(1)
         await GameMasterRequestor.endNightActionP(this._game.id, this._player.id)
+
         this._status = 'night action over'
         let game = await waitForStatusP(this._game.id, this._player.id, 'day')
         this._updateBoard(game)
         this._endNightP()
```

Comment:
You need to either await here or return this promise. Otherwise, the playP method will not be awaiting the resolution of this promise.